### PR TITLE
Feature/edit api

### PR DESCRIPTION
**Pull request by Feshchenko**
- split create api form to General and DataSource
- update General Api information
- update DataSource Api information
- code optimization

### CODE REVIEW

File: packages/frontend/src/presenter/ApiDefs/useUpdateDataSources.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    sources,
+    enabled,
+  } = api;
+  const { auth_header_name = '', auth_tokens = [] } = authentication || {};
+
+  const { data, loading } = useSources();
+  const { updateApiDef } = useUpdateApiDef({
+    onCompleted() {
+      refetch();
+      showSuccessToast(`Api ${name} successfully  updated`);
```
Comment (valcker) [inline packages/frontend/src/presenter/ApiDefs/useUpdateDataSources.ts:26]: `API ${name} successfully updated`

File: packages/frontend/src/view/MyAPI/Form/General.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  return (
+    <>
+      <FormGroup title="General settings">
+        <Row spacing={2}>
+          <Col xs={6}>
+            <Controller
+              as={Input}
+              name="name"
+              control={control}
+              label="Api name (required)"
```
Comment (valcker) [inline packages/frontend/src/view/MyAPI/Form/General.tsx:26]: API is an abbreviation, should be all-caps.

**Review by valcker — APPROVED** [review]

File: packages/frontend/src/presenter/ApiDefs/useUpdateDataSources.ts (hunk 1 above)
Comment (Feshchenko) [reply packages/frontend/src/presenter/ApiDefs/useUpdateDataSources.ts]: change to `API`

File: packages/frontend/src/view/MyAPI/Form/General.tsx (hunk 2 above)
Comment (Feshchenko) [reply packages/frontend/src/view/MyAPI/Form/General.tsx:26]: changed to `API ...`

**Pull request merged**